samza-core/src/main/java/org/apache/samza/clustermanager/ClusterBasedJobCoordinator.java

```diff
@@ -186,6 +209,18 @@ public void run() {
       //initialize JobCoordinator state
       log.info("Starting Cluster Based Job Coordinator");
 
+      //create necessary checkpoint and changelog streams, if not created
+      JobModel jobModel = jobModelManager.jobModel();
+      CheckpointManager checkpointManager = new TaskConfigJava(config).getCheckpointManager(metrics);
+      if (checkpointManager != null) {
+        checkpointManager.createStream();
+      }
+      changelogStreamManager.createChangeLogStreams(jobModel.getConfig(), jobModel.maxChangeLogStreamPartitions);
+
+      // Remap change
+      Map prevPartitionMappings = changelogStreamManager.readPartitionMapping();
+      changelogStreamManager.updatePartitionMapping(prevPartitionMappings, jobModel.getTaskPartitionMappings());
```

Comment:
QQ: Do you know why changelogStreamManager#updatePartitionMapping  doesn't read the mapping itself?